File: app/src/main/java/org/liberty/android/nc1020emu/NC1020JNI.java
package org.liberty.android.nc1020emu;

public class NC1020JNI {
	public static native void initialize(String path);
	public static native void reset();
	public static native void load();
	public static native void save();
	public static native void setKey(int keyId, boolean downOrUp);
	public static native void runTimeSlice(int timeSlice, boolean speedUp);
	public static native boolean copyLcdBuffer(byte[] buffer);
	public static native long getCycles();
	static {
		System.loadLibrary("nc1020");
	}
}

File: app/src/main/java/org/liberty/android/nc1020emu/NC1020JNI.kt
package org.liberty.android.nc1020emu

object NC1020JNI {
    @JvmStatic external fun initialize(romFilePath: String?, norFilePath: String?, stateFilePath: String?)
    @JvmStatic external fun reset()
    @JvmStatic external fun load()
    @JvmStatic external fun save()
    @JvmStatic external fun setKey(keyId: Int, downOrUp: Boolean)
    @JvmStatic external fun runTimeSlice(timeSlice: Int, speedUp: Boolean)
    @JvmStatic external fun copyLcdBufferEx(buffer: ByteArray?): Boolean
    @JvmStatic val cycles: Long external get

    init {
        System.loadLibrary("nc1020")
    }
}